hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ReplicationManager.java

```diff
@@ -785,6 +787,20 @@ private void sendDeleteCommand(final ContainerInfo container,
     tracker.accept(new InflightAction(datanode, Time.monotonicNow()));
   }
 
+  /**
+   * Wrap the call to nodeManager.getNodeStatus, catching any
+   * NodeNotFoundException and instead throwing a RuntimeException.
+   * @param dn The datanodeDetails to obtain the NodeStatus for
+   * @return NodeStatus corresponding to the given Datanode.
+   */
+  private NodeStatus getNodeStatus(DatanodeDetails dn) {
+    try {
+      return nodeManager.getNodeStatus(dn);
+    } catch (NodeNotFoundException e) {
+      throw new RuntimeException("Unable to find NodeStatus for "+dn, e);
```

Comment:
I think this should be more specific, eg. `IllegalStateException` seems to be a good candidate.  If you agree, please remember to update the corresponding `catch`, too.

Reply:
Yea that makes sense. This means that is a "real" runtime exception occurs, the catch will not stop it propagating up the stack and instead we only catch this more specific exception. I have made this change.